src/commonControls/listControl/ListControlAPL.ts

```diff
@@ -47,14 +51,16 @@ export namespace ListControlAPLPropsBuiltIns {
      * https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-data-source.html
      */
     export function TextListDataSourceGenerator(
-        slotIdMapper: { [index: string]: string } | ((choiceId: string) => string),
+        valueRenderer: (value: string, input: ControlInput) => string,
     ) {
-        return (act: RequestChangedValueByListAct) => {
+        return (act: RequestChangedValueByListAct, input: ControlInput, state: ControlState) => {
             const itemsArray: APLListItem[] = [];
             for (const choice of (act as any).payload.allChoices) {
                 itemsArray.push({
                     primaryText:
-                        typeof slotIdMapper === 'function' ? slotIdMapper(choice) : slotIdMapper[choice],
+                        typeof valueRenderer === 'function'
```

Comment:
the param type is now just function, so the ternary can be dropped I think.

Reply:
Updated